lib/on_generate_routes.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_form_exercise/screens/form/form_page.dart';
+
+class GenerateRoute {
+  static Route<dynamic>? generateRoute(RouteSettings settings) {
+    switch (settings.name) {
+      case ('/') : {
+        return MaterialPageRoute(
+          builder: (context) => const FormPage(),
+          settings: settings
+        );
+      }
+    }
+
+    return null;
```

Comment:
This is an error. You should `throw DeveloperError()` here. Define that error for yourself.

Reply:
Is it better to throw a custom defined RouteNotFound() here?

Reply:
Please let me clear my understanding. 
Keeping this code in example, it's evident that, the execution shouldn't reach that section of "return null". If due to some mistake in the underlying code, the code reaches that part, then that is a mistake made by me in the code part, which we are naming as a "Developer Error" as this behaviour is not expected.
But giving it a specific name (like  RouteNotFoundError) makes it look like it is already expected, which  is not the case here. 
Am I right on this @mayurdhaka-suryasoftware ?